---
title: "Introgression Analysis"
author: "Jinliang Yang"
date: "Jan. 5th, 2015"
output: rmarkdown::tufte_handout
---

# SUMMARY
Test whether Totontepec maize lines have admixture with _mexicana_ using SNP50k data.   

---------

# genetic and physical map of the SNP50k

I compared the SNP50k map from Matt's archive and the map downloaded from Illumina website. Matt's map is more inforamtive. It contains less SNPs (N=37,568) but more of them have physical information.

```{r, eval=FALSE}
source("../profiling/1.Introgression/1.A.1_snps_3popl.R")
```

# Phasing parental reference populations using __fastPHASE__

Run the following `R` script to format the SNP50k data and the formatted data were used as the `fastPHASE`[^r1] input. The formatted data included:  
1. Mexicana (N=120)  
2. Parviglumis (N=130)  
3. Landraces (N=94)   

```{r, eval=FALSE}
source("profiling/1.Introgression/1.A.1_fastPHASE_input.R")
```
And then run `fastPHASE` to do the haplotype phasing with [HMM](http://statweb.stanford.edu/~nzhang/Stat366/Lecture5.pdf). Basically, I ran the code, for example  
`fastPHASE -oMex -S1234 mex_120.fp`,   
to do the hap phasing. To run on `farm` with `slurm`, I prepared the slurm codes with an `R` script. 
```{r eval=FALSE}
source("../profiling/1.Introgression/1.A.2_run_fastPHASE.R")

# run the following code on farm
```
```
cd /home/jolyang/Documents/Github/Introgression
sbatch -p bigmeml largedata/fphase/mex_fp_slurm.sh
sbatch -p bigmeml largedata/fphase/parv_fp_slurm.sh
sbatch -p bigmeml largedata/fphase/land_fp_slurm.sh
```

# 


[^r1]: Scheet, P., & Stephens, M. (2006). A Fast and Flexible Statistical Model for Large-Scale Population Genotype Data : Applications to Inferring Missing Genotypes and Haplotypic Phase, 78(April), 629â€“644.


